pkg/sync/sync_service.go

```diff
@@ -121,12 +121,15 @@ func (syncService *SyncService[H]) initStoreAndStartSyncer(ctx context.Context,
 	if initial.IsZero() {
 		return errors.New("failed to initialize the store and start syncer")
 	}
-	if err := syncService.store.Init(ctx, initial); err != nil {
+
+	if err := syncService.StartSyncer(ctx); err != nil {
 		return err
 	}
-	if err := syncService.StartSyncer(ctx); err != nil {
+
+	if err := syncService.store.Append(ctx, initial); err != nil {
```

Comment:
Another potential gotcha with using `Append` that may affect the code here is that it is not synchronous, unlike `Init`, i.e., results of calling `Append` are not immediately reflected in `Height() or `Get/HasX` calls.  The temporary hack there is to use `Sync` method directly on the store (not on the interface.)

Reply:
done thanks!